packages/core/src/components/button/abstractButton.tsx

```diff
@@ -56,7 +56,10 @@ export interface IButtonState {
     isActive: boolean;
 }
 
-export abstract class AbstractButton<T> extends React.Component<React.HTMLProps<T> & IButtonProps, IButtonState> {
+export abstract class AbstractButton<H extends React.HTMLAttributes<any>> extends React.PureComponent<
```

Comment:
🤯

Reply:
emoji didn't come though, it's just a square. what's the sentiment here?

Reply:
![image](https://user-images.githubusercontent.com/199754/37621785-3c9e4b88-2b7d-11e8-84e0-5b0acd3e8f1b.png)

Reply:
oh yes very nice